crates/fluvio-stream-dispatcher/src/metadata/local.rs

```diff
@@ -0,0 +1,892 @@
+        let updates1: Vec<_> = updates1.into_iter().flatten().flatten().collect();
+        assert_eq!(updates1.len(), 3);
+
+        assert!(
+            matches!(updates1.get(0), Some(LSUpdate::Mod(obj)) if obj.status.to_string().eq(""))
+        );
+        assert!(
+            matches!(updates1.get(1), Some(LSUpdate::Mod(obj)) if obj.status.to_string().eq("new status"))
+        );
+        assert!(matches!(updates1.get(2), Some(LSUpdate::Delete(deleted)) if deleted.eq(&obj.key)));
+
+        let updates2: Vec<_> = updates2.into_iter().flatten().flatten().collect();
+        assert_eq!(updates2.len(), 0);
+    }
+
+    #[test]
+    fn test_ser() {
+        //given
+        let spec = TestSpec { replica: 1 };
+        let meta = LocalMetadataItem {
+            id: "meta1".to_string(),
+            revision: 1,
+        };
+        let parent = Some(LocalMetadataItem {
+            id: "parent1".to_string(),
+            revision: 2,
+        });
+
+        let spec_storage = SpecStorageV1 {
+            key: "key1".to_string(),
+            status: TestStatus("status1".to_string()),
+            spec,
+            meta,
+            parent,
+        };
+
+        //when
+        let str =
+            serde_yaml::to_string(&VersionedSpecStorage::V1(spec_storage)).expect("serialized");
+
+        //then
+        assert_eq!(
+            str,
+            r#"api-version: 1.0.0
+id: meta1
+revision: 1
+parent:
+  id: parent1
+  revision: 2
+key: key1
+status: status1
+spec:
+  replica: 1
+"#
+        );
+    }
+
+    #[test]
+    fn test_deser() {
+        //given
+        let input = r#"api-version: 1.0.0
+id: meta
+revision: 2
+parent:
+  id: parent1
+  revision: 3
+key: key1
+status: status3
+spec:
+  replica: 2
+"#;
+        //when
+        let parsed: VersionedSpecStorage<TestSpec> =
+            serde_yaml::from_str(input).expect("deserialized");
+
+        //then
+        assert_eq!(
+            parsed,
+            VersionedSpecStorage::V1(SpecStorageV1 {
+                meta: LocalMetadataItem {
+                    id: "meta".to_string(),
+                    revision: 2,
+                },
+                parent: Some(LocalMetadataItem {
+                    id: "parent1".to_string(),
+                    revision: 3,
+                }),
+                key: "key1".to_string(),
+                status: TestStatus("status3".to_string()),
+                spec: TestSpec { replica: 2 }
+            })
+        )
+    }
+
+    fn default_test_store_obj() -> LocalStoreObject<TestSpec> {
+        let meta = LocalMetadataItem {
+            id: "meta".to_string(),
+            revision: 0,
+        };
+        let parent = Some(LocalMetadataItem {
+            id: "parent".to_string(),
+            revision: 0,
+        });
+        let spec = TestSpec { replica: 1 };
+        LocalStoreObject::new_with_context(
+            meta.uid().to_string(),
+            spec,
+            MetadataContext::new(meta, parent),
+        )
+    }
```

Comment:
reducer is probably not place to implement it since it will duplicate logic or interfere with other metadata storage like K8.  that should be moved to metadata layer.

Anyway, it's good to see concurrency update tests.   It is good enough for this PR